### PR TITLE
CB-10679: Adding docs for new plugin version fetching implementation

### PR DESCRIPTION
**Pull request by riknoll**
Redo of old pull request that is rebased to master and clarifies a few things

### CODE REVIEW

**Comment by riknoll** [comment]
@nikhilkh I just put the Cordova version in bold in the first sentence of the section. Let me know if I should update that.

**Comment by riknoll** [comment]
Added blog post

File: www/docs/en/dev/guide/hybrid/plugins/index.md [hunk 1]
```diff
@@ -239,3 +239,105 @@ For example, for a plugin that supports android, iOS & Windows, the keywords in
...
+            DEPENDENCY: SEMVER_RANGE,
+            DEPENDENCY: SEMVER_RANGE,
+            ...
+        },
+        ...
+    }
+}
+```
+
+* `PLUGIN_VERSION` specifies a version of your plugin. It should adhere the syntax for a single version as defined by [npm's semver package][npm-semver] or an upper bound (see [below](#upper-bounds))
```
Comment (nikhilkh) [inline www/docs/en/dev/guide/hybrid/plugins/index.md]: Nit: You could as well hyper-link "upper bound"

**Comment by nikhilkh** [comment]
LGTM.

File: www/_posts/2016-03-16-new-plugin-fetching.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+The new API allows for plugin developers to give a mapping of plugin
+releases to project requirements in their `package.json`. When an app developer
+adds a plugin to their Cordova project, cordova-lib will examine their
+installed plugins, platforms, and cordova-lib version and decide which
+version of the plugin to fetch based on the provided mapping.
+If the project does not fulfill the requirements for the latest release of the
+plugin, the CLI will print warnings indicating the problems and fetch the
+latest release of the plugin that is compatible.
+
+This API may eventually replace the [engines element](http://cordova.apache.org/docs/en/latest/plugin_ref/spec.html#engines-and-engine) in `plugin.xml`.
```
Comment (stevengill) [inline www/_posts/2016-03-16-new-plugin-fetching.md]: I would say it **will** eventually replace the engine element in plugin.xml! But we can wait until we actually make that change

**Comment by stevengill** [comment]
LGTM

File: www/_posts/2016-03-16-new-plugin-fetching.md (hunk 2 above)
Comment (riknoll) [inline www/_posts/2016-03-16-new-plugin-fetching.md]: Sure!

**Comment by stevengill** [comment]
Lets merge and publish this! That way I can reference it in tools release blog post.

**Comment by riknoll** [comment]
What is the expected review time for blog posts? I sent an email to the dev list yesterday.

**Comment by stevengill** [comment]
We don't have a strict policy for blog posts. I think one day is good enough. We can always update blog posts at a later date if the need arises.

**Comment by riknoll** [comment]
Okay

**Pull request closed without merging**